primitives/src/marketplace.rs

```diff
@@ -36,30 +36,38 @@ pub enum MarketplaceType {
 #[derive(
 	Encode, Decode, CloneNoBound, Eq, PartialEqNoBound, RuntimeDebugNoBound, TypeInfo, MaxEncodedLen,
 )]
-#[scale_info(skip_type_params(AccountSizeLimit, OffchainDataLimit,))]
+#[scale_info(skip_type_params(AccountSizeLimit, OffchainDataLimit, CollectionSizeLimit))]
 #[codec(mel_bound(AccountId: MaxEncodedLen, Balance: MaxEncodedLen))]
-pub struct MarketplaceData<AccountId, Balance, AccountSizeLimit, OffchainDataLimit>
-where
+pub struct MarketplaceData<
+	AccountId,
+	Balance,
+	AccountSizeLimit,
+	OffchainDataLimit,
+	CollectionSizeLimit,
+> where
 	AccountId: Clone + PartialEq + Debug,
 	Balance: Clone + PartialEq + Debug + sp_std::cmp::PartialOrd,
 	AccountSizeLimit: Get<u32>,
 	OffchainDataLimit: Get<u32>,
+	CollectionSizeLimit: Get<u32>,
 {
 	pub owner: AccountId,
 	pub kind: MarketplaceType,
 	pub commission_fee: Option<CompoundFee<Balance>>,
 	pub listing_fee: Option<CompoundFee<Balance>>,
 	pub account_list: Option<BoundedVec<AccountId, AccountSizeLimit>>,
 	pub offchain_data: Option<U8BoundedVec<OffchainDataLimit>>,
+	pub collection_list: Option<BoundedVec<CollectionId, CollectionSizeLimit>>,
```

Comment:
Hmm, I think we made a mistake of having the `account_list` being a optional. What do you think?

Reply:
I think it's ok, it's not like it will make any problem in the code, even marketplaces won't use those feature so much.